### PR TITLE
HOLD: add meta set support

### PR DESCRIPTION
**Pull request by danmayer**
### TL;DR - What are you trying to accomplish?

This work tries to push the meta parser a bit further by adding meta set, and more specifically implementing part of meta set that would require it to parse the set response and return the meta flag response data to the caller.

### Details - How are you making this change?  What are the effects of this change?

This continues to work off the existing value object, but makes all fields in that response optional, so we can use it in a sparse response style. When using it with meta_set for example it will only contain the data requested with the set, for example the CAS value, opaque token, etc.

### TODO

- [x] rebase the main branch after set_multi langs and re-organize tests
- [ ] add benchmarks for meta_get and meta_set (benchmark write.await vs building command buffer)
- [x] look at defaults for meta_values
- [ ] refactor away more of the details around parsing meta flags
- [ ] add more tests for `parse_meta_tag_values_as_slice` and additional known flags

### CODE REVIEW

**Review by mrattle — APPROVED** [review]
Merging this onto the head branch and then doing some cleanup on there to get closer to reviewable first PR for the meta protocol.
